module/profile/helper.js

```diff
@@ -0,0 +1,134 @@
+// calling entities service for entity-managament
+const entityManagementService = require(GENERICS_FILES_PATH + '/services/entity-management')
+// calling user service api
+const userService = require(GENERICS_FILES_PATH + '/services/users')
+
+module.exports = class FormsHelper {
+	/**
+	 * @function read
+	 * @description Fetches and processes user profile data, including location details.
+	 * @param {Object} userId - The userId ccontaining the userId.
+	 * @returns {Promise<Object>} - A promise that resolves with the processed user profile data or an error object.
+	 * @throws {Error} - Throws an error if the user details cannot be fetched or processed.
+	 **/
+	static read(userId) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Fetch user profile details using userService.profile function
+				const userResponse = await userService.profile(userId)
+
+				// Check if the user profile fetch was successful
+				if (!userResponse.success) {
+					throw {
+						message: CONSTANTS.common.STATUS_FAILURE,
+						status: HTTP_STATUS_CODE.bad_request.status,
+					}
+				}
+				// Store the fetched user details
+				const userDetails = userResponse.data
+				if (!userDetails.meta) {
```

Comment:
@MallanagoudaB The aim was not to throw error if meta is not present. if meta is present and is not a empty object, perform locationId fetch and entity management calls and all. other wise return user details response after removing the location key